crates/noirc_frontend/src/util/mod.rs

```diff
@@ -0,0 +1,8 @@
+/// Equivalent to .iter().map(f).collect::<Vec<_>>()
+pub fn vecmap<T, U, F>(iterable: T, f: F) -> Vec<U>
```

Comment:
Since mapping an iterator is a common thing to do in Rust, I do wonder if this helper method is needed and what we would do if we needed something like `iterable.into_iter().map(f).filter(g).collect()` ?

Reply:
This helper method isn't strictly needed of course, I just added it to simplify some code that used that repeated pattern. If we need something more complex like your example then we'd just write it out like you did since there aren't enough repeated uses to warrant a helper for it.

Reply:
Yeah that makes sense, I don't have a strong opinion towards leaving it out, so will resolve this conversation